common/ledger/blockledger/fileledger/factory.go

```diff
@@ -50,22 +51,33 @@ func (f *fileLedgerFactory) GetOrCreate(channelID string) (blockledger.ReadWrite
 }
 
 // Remove removes an existing ledger and its indexes. This operation
-// is blocking.
+// performs an async deletion.
 func (f *fileLedgerFactory) Remove(channelID string) error {
 	f.mutex.Lock()
 	defer f.mutex.Unlock()
 
+	current, exists := f.toBeRemoved[channelID]
+	if exists && current {
+		return nil
+	}
+	f.toBeRemoved[channelID] = true
+
 	// check cache for open blockstore and, if one exists,
 	// shut it down in order to avoid resource contention
 	ledger, ok := f.ledgers[channelID]
 	if ok {
 		ledger.blockStore.Shutdown()
 	}
 
-	err := f.blkstorageProvider.Drop(channelID)
-	if err != nil {
-		return err
-	}
+	go func() {
```

Comment:
what happens if you crash during this procedure? does it make sense to somehow "mark" the channel for deletion synchronously, and then just asynchronously remove it?

Reply:
See previous comment.